src/Persistence/PersistenceManager.php

```diff
@@ -140,14 +143,53 @@ public function flush(ObjectManager $om): void
         }
     }
 
+    /**
+     * @template T of object
+     *
+     * @param T $object
+     */
+    public function autorefresh(object $object, object $clone): void
```

Comment:
I decided to split "refresh" and "autorefresh" mechanisms, because they does not involve the same problems, and the autorefresh should not manipulate the object reference (otherwise, we're exposed to identity problems)